File: cpp/selection_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

void printVector(vector<int>& arr) {
    for (int num : arr) {
        cout << num << " ";
    }  
    cout << endl;
}

void selectionSort(vector<int>& arr) {
    int size = arr.size();
    for (int i = 0; i < size - 1; i++) {
        int minIndex = i;
        
        for (int j = i + 1; j < size; j++) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }
        
        if (minIndex != i) {
            swap(arr[i], arr[minIndex]);
        }
        
        cout << "After pass " << i + 1 << ": ";
        printVector(arr);
    }
}

int main() {
    int size;
    cout << "Enter the size of array: ";
    cin >> size;

    vector<int> arr(size);
    cout << "Enter " << size << " elements: ";
    for (int i = 0; i < size; i++) {
        cin >> arr[i];
    }

    cout << "Original array: ";
    printVector(arr);

    cout << "\nSorting process:\n";
    selectionSort(arr);

    cout << "\nSorted array: ";
    printVector(arr);

    return 0;
}

File: python/selectionsort.py
def print_array(arr, size):
    for i in range(size):
        print(arr[i], end=" ")
    print()

def selection_sort(arr, size):
    for i in range(size - 1):
        min_index = i
        for j in range(i + 1, size):
            if arr[j] < arr[min_index]:
                min_index = j
        if min_index != i:
            temp = arr[i]
            arr[i] = arr[min_index]
            arr[min_index] = temp
        print_array(arr, size)


size = int(input("Enter the size of array: "))
arr = [0] * size 
    
print(f"Enter {size} elements: ", end="")
elements = input().split()
for i in range(size):
    arr[i] = int(elements[i])
    
print("Original array: ", end="")
print_array(arr, size)
    
selection_sort(arr, size)
    
print("Sorted array: ", end="")
print_array(arr, size)
